--- udftools-1.0.0b3/wrudf/wrudf.c.old	2012-01-20 22:01:27.540007203 +0400
+++ udftools-1.0.0b3/wrudf/wrudf.c	2012-01-20 22:02:13.142007362 +0400
@@ -245,8 +245,11 @@
 	    } else if( strncmp( spm->partIdent.ident, UDF_ID_VIRTUAL, strlen(UDF_ID_VIRTUAL)) == 0 )
 		virtualPartitionNum = i;
 	}
-	(char*)spm += spm->partitionMapLength;
-    }
+	//Ugly, but it compiles that way //Ncrmnt
+	char* u = (char*) spm;
+	u += spm->partitionMapLength;
+    spm = u;
+	}
 
     if( medium == CDR ) {
 	if( virtualPartitionNum != 0xFFFF )
